lib/modules/versioning/hermit/index.ts

```diff
@@ -0,0 +1,233 @@
+      return super.equals(version, other);
+    }
+    const parsedVersion = this._parse(version);
+    const parsedOther = this._parse(other);
+
+    if (parsedVersion !== null && parsedOther !== null) {
+      const versionReleases = parsedVersion.release;
+      const otherReleases = parsedOther.release;
+
+      if (versionReleases.length !== otherReleases.length) {
+        return false;
+      }
+
+      for (let i = 0; i < versionReleases.length; i++) {
+        if (versionReleases[i] !== otherReleases[i]) {
+          return false;
+        }
+      }
+
+      return true;
+    }
+
+    return version === other;
+  }
+
+  override isGreaterThan(version: string, other: string): boolean {
+    return this.sortVersions(version, other) > 0;
+  }
+
+  override isLessThanRange(version: string, range: string): boolean {
+    return !this.isGreaterThan(version, range);
+  }
+
+  private _filterVersions(versions: string[]): string[] {
+    return versions
+      .filter((v) => {
+        return this._isValidSemver(v) || this._isSemverChannel(v);
+      })
+      .map((v) => {
+        if (this._isSemverChannel(v)) {
+          return this._getChannel(v);
+        }
+
+        return v;
+      });
+  }
+
+  override getSatisfyingVersion(
+    versions: string[],
+    range: string
+  ): string | null {
+    return super.getSatisfyingVersion(this._filterVersions(versions), range);
+  }
+
+  override minSatisfyingVersion(
+    versions: string[],
+    range: string
+  ): string | null {
+    return super.minSatisfyingVersion(this._filterVersions(versions), range);
+  }
+
+  override sortVersions(version: string, other: string): number {
+    if (this._isValidSemver(version) && this._isValidSemver(other)) {
+      return super.sortVersions(version, other);
+    }
+
+    const parsedVersion = this._parse(version);
+    const parsedOther = this._parse(other);
+
+    if (parsedVersion === null || parsedOther === null) {
+      if (parsedVersion === null && parsedOther === null) {
+        return version.localeCompare(other);
+      }
+      return parsedVersion === null ? -1 : 1;
+    }
+
+    const versionReleases = parsedVersion.release;
+    const otherReleases = parsedOther.release;
+
+    const maxLength =
+      versionReleases.length > otherReleases.length
+        ? versionReleases.length
+        : otherReleases.length;
+
+    for (let i = 0; i < maxLength; i++) {
+      const verVal = versionReleases[i];
+      const otherVal = otherReleases[i];
+
+      if (
+        verVal !== undefined &&
+        otherVal !== undefined &&
+        verVal !== otherVal
+      ) {
+        return verVal - otherVal;
+      } else if (verVal === undefined) {
+        return 1;
+      } else if (otherVal === undefined) {
+        return -1;
+      }
+    }
+
+    return 0;
+  }
+
+  override matches(version: string, range: string): boolean {
+    return this.equals(version, range);
+  }
+}
+
+export const api: VersioningApiConstructor = HermitVersioning;
```

Comment:
Not sure if you have to new up the class here or not 🤔

Reply:
I'm following the examples in RegExpVersioningApi. It doesn't look like I need it to be newed up here.

https://github.com/renovatebot/renovate/blob/main/lib/modules/versioning/regex/index.ts#L155